packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_global_client.test.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { loggingSystemMock } from '@kbn/core-logging-server-mocks';
+import { UiSettingsParams } from '@kbn/core-ui-settings-common';
+import { savedObjectsClientMock } from '@kbn/core-saved-objects-api-server-mocks';
+import { UiSettingsGlobalClient } from './ui_settings_global_client';
+
+const logger = loggingSystemMock.create().get();
+
+const TYPE = 'config-global';
+const ID = 'kibana-version';
+const BUILD_NUM = 1234;
+
+interface SetupOptions {
+  defaults?: Record<string, any>;
+  esDocSource?: Record<string, any>;
+  overrides?: Record<string, any>;
+}
+
+describe('ui settings global client', () => {
+  const settingA: UiSettingsParams = {
+    name: 'settingA',
+    value: 'abc',
+    scope: 'global',
+    schema: schema.string(),
+  };
+  const settingB: UiSettingsParams = {
+    name: 'settingA',
+    value: 123,
+    scope: 'global',
+    schema: schema.number(),
+  };
```

Comment:
Is it normal that both `settingA` and `settingB` are `name: "settingA"`? If not, I think we're missing a check in setting registration, as this should have thrown?

Reply:
Good catch! I don't think it is supposed to throw, as it will just override the setting value? The question is - should we throw if `defaults` contains multiple settings with the same name?

Reply:
Oh sorry these were the `defaults`... I guess we can assume that default values won't have dupes